opal/corelib/kernel.rb

```diff
@@ -779,16 +779,33 @@ def instance_variable_get(name)
   def instance_variable_set(name, value)
     name = Opal.instance_variable_name!(name)
 
-    `self[name.substr(1)] = value`
+    `self[Opal.ivar(name.substr(1))] = value`
   end
 
+  %x{
+    function restoreIvar(name) {
+      if (name === "constructor$" ||
+        name === "__proto__$" ||
+        name === "__parent__$" ||
+        name === "__noSuchMethod__$" ||
+        name === "__count__$" ||
+        name === "hasOwnProperty$" ||
+        name === "valueOf$")
+      {
+        return name.slice(0, name.length - 1);
+      }
+
+      return name;
+    }
+  }
+
```

Comment:
@elia you mean remove all the `===`'s or just instead of `name.slice(...)`? In case it's the latter, wouldn't `name.slice` be more efficient than doing a regexp?